### DISCUSSION TITLE
Channel Estimation in Ray Tracing Scenario

### DISCUSSION THREAD

**Discussion by Priyanshjain24** [NVlabs/sionna#523] [Q&A]
Hi. I am using the `simple_street_canyon_with_cars` scenario in ray tracing.. I have setup a transmitter (on top of a moving car) and a receiver (stationary) in the scenario. I want to find out the real and the estimated channel between the transmitter and receiver using OFDM communication. To find out the real channel, I used the `compute_paths`, `apply_doppler` and `cir_to_ofdm_channel` functions. But I am unable to figure out how to do the estimation of the channel in the ray tracing scenario using OFDM communication. Any help would be appreciated. Thanks.

**Comment by SebastianCa** [comment]
Hi @Priyanshjain24,

Please refer to the [Ray Tracing Tutorial](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Introduction.html#Site-specific-Link-Level-Simulations). Specifically, the section on site-specific link-level simulations explains how to conduct end-to-end link-level simulations. Additionally, an example of channel estimation is provided in [this notebook](https://nvlabs.github.io/sionna/examples/MIMO_OFDM_Transmissions_over_CDL.html#Uplink-Transmission-in-the-Frequency-Domain).

If you are still facing issues, could you please share a code snippet that illustrates where you are conceptually stuck?

**Comment by Priyanshjain24** [comment]
Hi @SebastianCa. Thank you for sharing the links. In the first tutorial, I was not able to understand why we are creating a coverage map and calculating paths when we want to do the estimation. For estimation, we would not know how far or how with what velocity an object is moving right? We would need to rely on some allocation of the resource grid to figure out that the received data is from which user. Also in that case, how would multipath propogation be handled? From the second tutorial, I was not able to understand how to link the mapper and demapper to transmitter and recievers in raytracing scenarios. Also, I was not able to understand how to send our own data (even testing binary data) using these transmitters in the ray tracing scenarios.

**Reply by priyapai21** [reply]
Hi @Priyanshjain24  I’m also working on the same and facing similar confusion. Did you get any clarity on it?

**Reply by jhoydis** [reply]
Hey @priyapai21 

What's your exact problem here? Can you rephrase it? The question you refer to is not super clear to me.

**Reply by priyapai21** [reply]
Hi @jhoydis Hi, I’m working on the same setup and had a doubt about channel estimation. Could you please share how you approached channel estimation in this case and which channel is actually estimated ?

Also, how to approach  the deep learning part—especially the CNN-based channel estimation? I’m a bit confused about how to train the model and what inputs/outputs to use. which dataset is suitable wt features should be included in it ?

Would really appreciate your guidance!

**Reply by jhoydis** [reply]
There is not difference in channel estimation between the case where the channels are simulated and when they are ray traced.
Have a look at this notebook https://nvlabs.github.io/sionna/phy/tutorials/notebooks/Realistic_Multiuser_MIMO_Simulations.html.
The channel estimator is specifically called out. 

Your question about CNN-based channel estimation seems not related to this and is rather a general question about ML for the PHY. Maybe ask this question in a detailed discussion item, providing more context about what you would like to accomplish.

**Reply by priyapai21** [reply]
@jhoydis thanks for sharing this . I had a quick doubt while going through the results. Why are we plotting BLER vs Eb/No instead of BER vs Eb/No in this setup?

Is there a specific reason for preferring BLER here (like coding or system-level evaluation)? Would appreciate your clarification!

**Comment by SebastianCa** [comment]
In the tutorial, we sample random user positions from the coverage to ensure that each position fulfils certain criteria (e.g., max/min path gains). Alternatively, you could also provide your own custom ue position (e.g., by placing them manually).
An example of how to run link-level simulations for ray traced channels is given [here](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Introduction.html#Site-specifc-Link-Level-Simulations). To send your own data, you need to set the flag `return_bits=False` in the PUSCHTransmitter (see [here](https://nvlabs.github.io/sionna/api/nr.html#puschtransmitter)) and feed your custom payload to the transmitter.
